tests/e2e/datadog/metrics/series.rs

```diff
@@ -0,0 +1,278 @@
+            vector::common::datadog::DatadogMetricType::Rate => 2,
+        };
+
+        MetricSeries {
+            resources,
+            metric: input.metric.clone(),
+            tags: input.tags.clone().unwrap_or_default(),
+            points,
+            r#type,
+            unit: "".to_string(),
+            source_type_name: input.clone().source_type_name.unwrap_or_default(),
+            interval,
+            metadata: None,
+        }
+    }
+}
+
+fn convert_v1_payloads_v2(input: &[DatadogSeriesMetric]) -> Vec<MetricPayload> {
+    input
+        .iter()
+        .map(|serie| MetricPayload {
+            series: vec![serie.into()],
+        })
+        .collect()
+}
+
+fn unpack_v1_series(in_payloads: &[FakeIntakePayloadJson]) -> Vec<DatadogSeriesMetric> {
+    in_payloads
+        .iter()
+        .flat_map(|payload| {
+            let series = payload.data.as_array().unwrap();
+            series
+                .iter()
+                .map(|serie| serde_json::from_value(serie.clone()).unwrap())
+        })
+        .collect()
+}
+
+async fn get_v1_series_from_pipeline(address: String) -> SeriesIntake {
+    info!("getting v1 series payloads");
+    let payloads =
+        get_fakeintake_payloads::<FakeIntakeResponseJson>(&address, SERIES_ENDPOINT_V1).await;
+
+    info!("unpacking payloads");
+    let payloads = unpack_v1_series(&payloads.payloads);
+    info!("converting payloads");
+    let payloads = convert_v1_payloads_v2(&payloads);
+
+    info!("generating series intake");
+    let intake = generate_series_intake(&payloads);
+
+    common_series_assertions(&intake);
+
+    info!("{:?}", intake);
+
+    intake
+}
+
+async fn get_v2_series_from_pipeline(address: String) -> SeriesIntake {
+    info!("getting v2 series payloads");
+    let payloads =
+        get_fakeintake_payloads::<FakeIntakeResponseRaw>(&address, SERIES_ENDPOINT_V2).await;
+
+    info!("unpacking payloads");
+    let payloads = unpack_proto_payloads::<MetricPayload>(&payloads);
+
+    info!("generating series intake");
+    let intake = generate_series_intake(&payloads);
+
+    common_series_assertions(&intake);
+
+    info!("{:?}", intake);
+
+    intake
+}
+
+pub(super) async fn validate() {
+    info!("==== getting series data from agent-only pipeline ==== ");
+    let agent_intake = get_v2_series_from_pipeline(fake_intake_agent_address()).await;
+
+    info!("==== getting series data from agent-vector pipeline ====");
+    let vector_intake = get_v1_series_from_pipeline(fake_intake_vector_address()).await;
+
+    assert_eq!(
+        agent_intake.len(),
+        vector_intake.len(),
+        "different number of unique Series contexts received"
+    );
+
+    agent_intake
+        .iter()
+        .zip(vector_intake.iter())
+        .for_each(|(agent_ts, vector_ts)| {
+            assert_eq!(agent_ts.0, vector_ts.0, "Mismatch of series context");
+
+            let metric_type = agent_ts.0.r#type;
+
+            // gauge: last one wins.
```

Comment:
Would it be helpful to fail here if the metric type is not gauge or rate? Otherwise, we would silently pass here without making any assertions.

Reply:
This is validated earlier on in the code path.

In `common_series_assertions()` , we validate that each metric type that we expect, is received.

In `generate_series_intake()`, we massage the data into a structure that maps the series context to the data. Part of the series context is the metric type.

On line 237, we confirm the contexts are the same, which would fail if the type does not match between the Agent/Agent-Vector case.

Reply:
However ... the submission vs in-app type is implicit in the validation. Because we are using dogstatsd in the emitter, the type should never be count. I see the value in your suggestion, I'll add that.